src/H5FScache.c

```diff
@@ -1231,7 +1231,9 @@ H5FS__cache_sinfo_serialize(const H5F_t *f, void *_image, size_t len, void *_thi
     /* Compute checksum */
 
     /* There may be empty space between entries and chksum */
-    chksum_image    = (uint8_t *)(_image) + len - H5FS_SIZEOF_CHKSUM;
+    chksum_image = (uint8_t *)(_image) + len - H5FS_SIZEOF_CHKSUM;
+    /* Reset the unused space between entries and chksum */
+    memset(image, 0, len - H5FS_SIZEOF_CHKSUM - ((void *)image - _image));
```

Comment:
I'm not opposed to merging this in, but I feel like it's better to just memset the entire buffer and avoid the pointer math.